mp/src/game/shared/neo/weapons/weapon_neobasecombatweapon.cpp

```diff
@@ -500,12 +500,16 @@ void CNEOBaseCombatWeapon::ProcessAnimationEvents()
 	else if (m_bLowered && m_bRoundBeingChambered)
 	{ // For bolt action weapons
 		m_bLowered = false;
-		next(ACT_VM_PULLBACK);
+		SendWeaponAnim(ACT_VM_PULLBACK);
+		m_flNextPrimaryAttack = max(gpGlobals->curtime + 1.2f, m_flNextPrimaryAttack);
+		m_flNextSecondaryAttack = m_flNextPrimaryAttack;
```

Comment:
We could maybe refactor this such that all the paths use the same `next` lambda for setting the values:

```diff
diff --git a/mp/src/game/shared/neo/weapons/weapon_neobasecombatweapon.cpp b/mp/src/game/shared/neo/weapons/weapon_neobasecombatweapon.cpp
index b90b0cec..dd0e669a 100644
--- a/mp/src/game/shared/neo/weapons/weapon_neobasecombatweapon.cpp
+++ b/mp/src/game/shared/neo/weapons/weapon_neobasecombatweapon.cpp
@@ -475,13 +475,12 @@ void CNEOBaseCombatWeapon::ProcessAnimationEvents()
                return;
        }
 
-       const auto next = [this](const int activity) {
+       const auto next = [this](const int activity, const float nextAttackDelay = 0.2) {
                SendWeaponAnim(activity);
                if (GetNeoWepBits() & NEO_WEP_THROWABLE)
                {
                        return;
                }
-               constexpr auto nextAttackDelay = 0.2;
                m_flNextPrimaryAttack = max(gpGlobals->curtime + nextAttackDelay, m_flNextPrimaryAttack);
                m_flNextSecondaryAttack = m_flNextPrimaryAttack;
        };
@@ -500,9 +499,7 @@ void CNEOBaseCombatWeapon::ProcessAnimationEvents()
        else if (m_bLowered && m_bRoundBeingChambered)
        { // For bolt action weapons
                m_bLowered = false;
-               SendWeaponAnim(ACT_VM_PULLBACK);
-               m_flNextPrimaryAttack = max(gpGlobals->curtime + 1.2f, m_flNextPrimaryAttack);
-               m_flNextSecondaryAttack = m_flNextPrimaryAttack;
+               next(ACT_VM_PULLBACK, 1.2f);
        }
```